metadata/metadata.go

```diff
@@ -53,3 +62,148 @@ func (m MetadataHeaderInfo) Validate() error {
 	}
 	return nil
 }
+
+var ErrInvalidTypeInfo = errors.New("invalid type info")
+
+type MetadataTypeInfo struct {
+	Rootfs string `json:"rootfs"`
+}
+
+func (m MetadataTypeInfo) Validate() error {
+	if len(m.Rootfs) == 0 {
+		return ErrInvalidTypeInfo
+	}
+	return nil
+}
+
+var ErrInvalidMetadata = errors.New("invalid metadata")
+
+type Metadata struct {
+	// we don't know exactly what tyoe of data we will have here
+	data map[string]interface{}
+}
+
+func (m Metadata) Validate() error {
+	if m.data == nil {
+		return ErrInvalidMetadata
+	}
+	// mandatory fields
+	var deviceType interface{}
+	var imageID interface{}
+
+	for k, v := range m.data {
+		if v == nil {
+			return ErrInvalidMetadata
+		}
+		if strings.Compare(k, "DeviceType") == 0 {
+			deviceType = v
+		}
+		if strings.Compare(k, "ImageID") == 0 {
+			imageID = v
+		}
+	}
+	if deviceType == nil || imageID == nil {
+		return ErrInvalidMetadata
+	}
+	return nil
+}
+
+var ErrInvalidFilesInfo = errors.New("invalid files info")
+
+type MetadataFile struct {
+	File string `json:"file"`
```

Comment:
Right. I think we need to change documentation. It states:
```
[
  "core-image-minimal-201608110900.ext4"
]
```
but this is not valid JSON.